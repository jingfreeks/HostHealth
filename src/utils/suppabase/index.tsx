export {default as supabase} from './suppabase';
