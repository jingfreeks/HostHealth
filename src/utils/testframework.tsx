import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
