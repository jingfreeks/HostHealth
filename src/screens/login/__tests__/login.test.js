import React from 'react';
import {render} from '@testing-library/react-native';
import {MockProvider} from '@/utils/testframework';
import Login from '../login';

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      navigate: () => {},
    }),
    useIsFocused: () => true,
    useDispatch: () => ({dispatch: jest.fn()}),
  };
});
describe('Login Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = render(
      <MockProvider
        store={{
          pcities: {loading: false, data: []},
          suggetedjob: {loading: false, data: []},
        }}>
        <Login />
      </MockProvider>,
    );
    expect(all.toJSON()).toMatchSnapshot();
  });
});
