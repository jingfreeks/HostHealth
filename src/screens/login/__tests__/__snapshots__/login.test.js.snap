// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Should work as expected to get snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ],
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <RCTSafeAreaView
      style={
        [
          {},
        ]
      }
    >
      <View
        alignType="center"
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "column",
            },
          ]
        }
      >
        <RCTSafeAreaView
          style={
            [
              {},
            ]
          }
        >
          <View
            alignType="center"
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 58.85294117647059,
                    "marginTop": 98.08823529411764,
                  },
                ]
              }
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/mainlogo.png",
                  }
                }
                style={
                  [
                    {
                      "height": 196.17647058823528,
                      "width": 451.20588235294116,
                    },
                  ]
                }
              />
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </RCTSafeAreaView>
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <RCTSafeAreaView
          style={
            [
              {},
            ]
          }
        >
          <View
            alignType="flex-start"
            style={
              [
                {
                  "alignItems": "flex-start",
                  "alignSelf": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 18,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Email
            </Text>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                onChangeText={[Function]}
                placeholder="Email"
                secureTextEntry={false}
                style={
                  [
                    {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 30,
                      "marginBottom": 5,
                    },
                  ]
                }
                value=""
              />
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#C2D6DC",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignSelf": "stretch",
                    "borderColor": "#C2D6DC",
                    "borderWidth": 0.5,
                    "flexDirection": "row",
                    "marginBottom": 15,
                  },
                ]
              }
            />
          </View>
        </RCTSafeAreaView>
        <RCTSafeAreaView
          style={
            [
              {},
            ]
          }
        >
          <View
            alignType="flex-start"
            style={
              [
                {
                  "alignItems": "flex-start",
                  "alignSelf": "flex-start",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#0e3f6c",
                    "fontFamily": "Avenir",
                    "fontSize": 18,
                    "fontWeight": "900",
                  },
                ]
              }
            >
              Password
            </Text>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                onChangeText={[Function]}
                placeholder="Password"
                secureTextEntry={false}
                style={
                  [
                    {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "height": 30,
                      "marginBottom": 5,
                    },
                  ]
                }
                value=""
              />
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#C2D6DC",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignSelf": "stretch",
                    "borderColor": "#C2D6DC",
                    "borderWidth": 0.5,
                    "flexDirection": "row",
                    "marginBottom": 15,
                  },
                ]
              }
            />
          </View>
        </RCTSafeAreaView>
      </View>
      <View
        style={
          [
            {
              "marginLeft": 5,
              "marginRight": 5,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            },
          ]
        }
      >
        <View
          border={10}
          borderc="#99E1E1"
          borderw={2}
          style={
            [
              {
                "borderColor": "#99E1E1",
                "borderRadius": 10,
                "borderWidth": 2,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#d6f3f3",
                "borderRadius": 10,
                "opacity": 1,
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#03606A",
                    "fontFamily": "Avenir",
                    "fontSize": 15,
                    "fontWeight": "500",
                  },
                ]
              }
            >
              SIGN IN
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "marginTop": 39.23529411764706,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          Forgot Password?
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 58.85294117647059,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          Forgot Password?
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginLeft": 5,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#0e3f6c",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Sign Up here
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
